src/web/extractors.rs

```diff
@@ -863,6 +863,30 @@ impl FromRequest for ConfigRequest {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct HeartbeatRequest {
+    pub headers: HeaderMap,
+    pub db: Box<dyn Db>,
+}
+
+impl FromRequest for HeartbeatRequest {
+    type Config = ();
+    type Error = Error;
+    type Future = Box<dyn Future<Item = Self, Error = Self::Error>>;
+
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        let headers = req.headers().clone();
+
+        let state = req.app_data::<ServerState>().unwrap();
```

Comment:
not terribly important but let's avoid this unwrap. You can copy one of the other ServerState gets in this file using a match statement so it can return a `future::err`,